docs/building-a-form/about-the-json-schema-standard-and-react-jsonschema-form.md

```diff
@@ -0,0 +1,296 @@
+```
+{
+  type: 'array',
+  items: {
+    type: 'object',
+    properties: {
+      field1: {
+        type: 'string'
+      },
+      field2: {
+        type: 'string'
+      }
+    }
+  }
+}
+```
+
+##### Field components
+
+These field components pass a collection of props down through each component. Most are passed to widget components.
+
+- `name`: The property name of the current field. For example, the object schema above would be named `field1`.
+- `required`: If the field is required or not (i.e. the property name is in the schema's `required` array).
+- `schema`: The schema for the specific field.
+- `uiSchema`: The ui schema for this field. See "[About the `uiSchema` field component](#about-the-uischema-field-component)."
+- `errorSchema`: An object that contains the list of errors for the current field and any child properties, if the field is an array or object.
+- `idSchema`: An object that contains the field IDs for the current field and any child properties. The library generates IDs for each field by joining each property name with an underscore.
+- `formData`: The actual data entered for the field so far.
+- `onChange`: The function that's called when data changes. See "[About the `onChange` field component](#about-the-onchange-field-component)."
+- `onBlur`: The function that's called when focus is lost on a widget.
+
+##### About the `onChange` field component
+
+When a user enters data, each widget calls `onChange`. Each component in the hierarchy passes an `onChange` handler to child fields. When child data changes, the component combines it with the rest of the data and calls the `onChange` prop passed to it from its parent.
+
+```
+{
+  type: 'object',
+  properties: {
+    field1: {
+      type: 'string'
+    }
+  }
+}
+```
+
+In this example:
+
+1. The user types 'a'.
+2. The `TextWidget` for field1 calls `onChange` with 'a'.
+3. The `onChange` property came from the parent `ObjectField` component, which puts 'a' in an object as `field1` (`{ field1: 'a' }`), then calls the `onChange` prop it was passed.
+4. When it reaches the top-level `Form` component, rjsf runs the JSON Schema validation and passes the results through the component hierarchy.
+
+Similar to Redux, all state is kept in the `Form` component, or the root of the form. All data processing and validation happens in `Form`, or is triggered by hooks provided by `Form`. The schemaform code built on top of this processes the schemas and form data in Redux, triggered by events provided by `Form`.
+
+##### About the `uiSchema` field component
+
+Along with the regular JSON Schema, a UI schema for UI-specific options that don't fit within the JSON Schema standard is also optionally defined for each field. The `uiSchema` field component is primarily used for specifying custom fields and widgets for specific fields in the schema using `ui:field` and `ui:widget`. In the schemaform, it's also used for label names, custom validation, and conditionally required fields.
```

Comment:
This discussion is pretty thin, I can't tell from the description how to create one and there isn't an example. Maybe we could add a todo for this to get its own page with more info eventually?